frontend/appflowy_flutter/lib/plugins/database/grid/presentation/widgets/sort/create_sort_list.dart

```diff
@@ -15,97 +14,56 @@ import 'package:flowy_infra_ui/widget/spacing.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
-class GridCreateSortList extends StatefulWidget {
-  const GridCreateSortList({
+class CreateDatabaseViewSortList extends StatelessWidget {
+  const CreateDatabaseViewSortList({
     super.key,
-    required this.viewId,
-    required this.fieldController,
-    required this.onClosed,
-    this.onCreateSort,
+    required this.onTap,
   });
 
-  final String viewId;
-  final FieldController fieldController;
-  final VoidCallback onClosed;
-  final VoidCallback? onCreateSort;
-
-  @override
-  State<StatefulWidget> createState() => _GridCreateSortListState();
-}
-
-class _GridCreateSortListState extends State<GridCreateSortList> {
-  late CreateSortBloc editBloc;
-
-  @override
-  void initState() {
-    editBloc = CreateSortBloc(
-      viewId: widget.viewId,
-      fieldController: widget.fieldController,
-    )..add(const CreateSortEvent.initial());
-    super.initState();
-  }
+  final VoidCallback onTap;
 
   @override
   Widget build(BuildContext context) {
-    return BlocProvider.value(
-      value: editBloc,
-      child: BlocListener<CreateSortBloc, CreateSortState>(
-        listener: (context, state) {
-          if (state.didCreateSort) {
-            widget.onClosed();
-          }
-        },
-        child: BlocBuilder<CreateSortBloc, CreateSortState>(
-          builder: (context, state) {
-            final cells = state.creatableFields.map((fieldInfo) {
-              return SizedBox(
-                height: GridSize.popoverItemHeight,
-                child: GridSortPropertyCell(
-                  fieldInfo: fieldInfo,
-                  onTap: (fieldInfo) => createSort(fieldInfo),
-                ),
-              );
-            }).toList();
-
-            final List<Widget> slivers = [
-              SliverPersistentHeader(
-                pinned: true,
-                delegate: _SortTextFieldDelegate(),
-              ),
-              SliverToBoxAdapter(
-                child: ListView.separated(
-                  shrinkWrap: true,
-                  itemCount: cells.length,
-                  itemBuilder: (BuildContext context, int index) {
-                    return cells[index];
-                  },
-                  separatorBuilder: (BuildContext context, int index) {
-                    return VSpace(GridSize.typeOptionSeparatorHeight);
-                  },
-                ),
-              ),
-            ];
-            return CustomScrollView(
+    return BlocBuilder<SortEditorBloc, SortEditorState>(
+      builder: (context, state) {
+        final filter = state.filter.toLowerCase();
+        final cells = state.creatableFields
+            .where((field) => field.field.name.toLowerCase().contains(filter))
```

Comment:
a quick note: if the names in the field are the same as the filter's name, there's a function called `equalsIgnoreCase` for case-insensitive comparison

Reply:
Oh cool! I did not know that. I will put it in the newest PR.